blocks/color-palette/index.js

```diff
@@ -42,38 +55,43 @@ class ColorPalette extends Component {
 				{ colors.map( ( color ) => {
 					const style = { color: color };
 					const className = classnames( 'blocks-color-palette__item', { 'is-active': value === color } );
-					/* Disable reason: aria-current seems like a valid aria-prop and well-suited for this */
-					/* eslint-disable jsx-a11y/aria-props */
+
 					return (
 						<div key={ color } className="blocks-color-palette__item-wrapper">
 							<button
+								type="button"
 								className={ className }
 								style={ style }
 								onClick={ () => onChange( value === color ? undefined : color ) }
 								aria-label={ sprintf( __( 'Color: %s' ), color ) }
-								aria-current={ value === color && __( 'Selected color' ) }
+								aria-pressed={ value === color ? true : false }
 							/>
 						</div>
 					);
-					/* eslint-enable jsx-a11y/aria-props */
 				} ) }
 
 				<div className="blocks-color-palette__item-wrapper blocks-color-palette__custom-color">
 					<button
+						type="button"
+						aria-expanded={ this.state.opened }
 						className="blocks-color-palette__item"
-						onClick={ this.openPicker }
-						aria-label={ __( 'Open custom color picker' ) }
-					/>
+						onClick={ this.togglePicker }
+						ref={ this.bindToggleNode }
+						aria-label={ __( 'Custom color picker' ) }
+					>
+						<span className="blocks-color-palette__custom-color-gradient"></span>
+					</button>
 					<Popover
 						isOpen={ this.state.opened }
-						onClose={ this.closePicker }
+						onClose={ this.closeOnClickOutside }
```

Comment:
As of deb86a78f14196dfe20fa5297a3a7970dd82a8b8, `Popover`'s `onClose` callback no longer passes the click `event`, since there can be many reasons a Popover might emit a close intent. The change here should be simply to change `onClose={` to `onClickOutside={`

I expect you may need to `git rebase origin/master` and force push the branch for this to take effect, since #2697 was merged to master after your branch was created.